File: src/lib/scala/Slate.Common/src/main/scala/slate/common/encrypt/EncryptSupport.scala
/**
<slate_header>
  author: Kishore Reddy
  url: https://github.com/kishorereddy/scala-slate
  copyright: 2015 Kishore Reddy
  license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  desc: a scala micro-framework
  usage: Please refer to license on github for more info.
</slate_header>
  */

package slate.common.encrypt

trait EncryptSupport {

  val encryptor: Option[Encryptor] = None


  def encrypt(text: String): String = {
    val result = encryptor.flatMap[String]( e => Some(e.encrypt(text)))
    result.getOrElse(text)
  }


  def decrypt(encrypted: String): String = {
    val result = encryptor.flatMap[String]( e => Some(e.decrypt(encrypted)))
    result.getOrElse(encrypted)
  }
}

File: src/common/common/src/main/kotlin/kiit/common/crypto/EncryptSupport.kt
/**
 *  <kiit_header>
 * url: www.kiit.dev
 * git: www.github.com/slatekit/kiit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * 
 *
  *  </kiit_header>
 */

package kiit.common.crypto

import kiit.common.Ignore

interface EncryptSupport {

    val encryptor: Encryptor?

    @Ignore
    fun encrypt(text: String): String {
        val result = encryptor?.let { e -> e.encrypt(text) }
        return result ?: text
    }

    @Ignore
    fun decrypt(encrypted: String): String {
        val result = encryptor?.let { e -> e.decrypt(encrypted) }
        return result ?: encrypted
    }
}
